model/riscv_sys_regs.sail

```diff
@@ -302,6 +310,9 @@ function legalize_mstatus(o : Mstatus, v : xlenbits) -> Mstatus = {
    */
   let m : Mstatus = Mk_Mstatus(zero_extend(v[22 .. 7] @ 0b0 @ v[5 .. 3] @ 0b0 @ v[1 .. 0]));
 
+  /* Legalize MPP */
+  let m = update_MPP(m, legalize_priv_level(m.MPP()));
```

Comment:
```suggestion
  let m = [m with MPP = legalize_priv_level(m[MPP])];
```

Reply:
This is nicer. Had not seen it used elsewhere. Is this new supported syntax?

Reply:
Yes I was also not aware of it until #373 but apparently it's been supported for quite a long time.